### ISSUE TITLE
Error on OctoberCMS 2.-

### ISSUE THREAD

**Issue by MateThemes** [OFFLINE-GmbH/oc-mall-plugin#751]
Hello,

i have an error when I want to add a product:

`"Argument 2 passed to OFFLINE\Mall\Models\Product::OFFLINE\Mall\Models\{closure}() must be of the type array, null given" on line 246 of /Users/geraldschandl/Sites/octobernew/plugins/offline/mall/models/Product.php`

What does this error mean?

**Comment by tobias-kuendig** [comment]
Can you check if this fixes this problem for you:

https://github.com/OFFLINE-GmbH/oc-mall-plugin/blob/develop/models/Product.php#L246

```diff

-        $this->bindEvent('model.relation.beforeDetach', function (string $relationName, array $attachedIdList) {
+        $this->bindEvent('model.relation.beforeDetach', function ($relationName, $attachedIdList) {
```

This might be PHP version related

**Comment by MateThemes** [comment]
Hello!

Yeah, this fixed the problem!

Thank you

**Comment by MateThemes** [comment]
I also need to change line 243 to save the category. Since the update there are some troubles.

**Comment by MateThemes** [comment]
`public function __construct($attributes = [])
    {
        parent::__construct($attributes);
        $this->bindEvent('model.relation.beforeAttach',
            function ($relationName, $attachedIdList, $insertData) {
                $this->forceReindex = true;
            });
        $this->bindEvent('model.relation.beforeDetach', function ($relationName, $attachedIdList) {
            $this->forceReindex = true;
        });
    }`

**Comment by tobias-kuendig** [comment]
Thanks for this bug report @MateThemes! This issue is fixed via 31400f8, released as v1.13.5
